### PR TITLE
add HazelcastCloudDiscovery code sample

### LINKED ISSUE #318: Hazelcast Cloud Discovery Configurations
The implementation will be out with 3.10.3 and 3.11
But we want to document this feature when hazelcast.cloud is out. So adding this to referance manual will wait for the hazelcast.cloud release. 

Note that the configurations are different in 3.10.3 and 3.11.

### LINKED ISSUE #319: Hazelcast Cloud Discovery Sample
We need to add a code sample for cloud discovery when hazelcast.cloud is out.

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This pr adds HazelcastCloudDiscovery feature to `README.md` and its configuration to `CONFIG.md`. It also adds a sample usage of HazelcastCloudDiscovery.

 fixes #318 
 fixes #319

### CODE REVIEW

File: CONFIG.md [hunk 1]
```diff
@@ -291,6 +291,30 @@ the properties section in the JSON configuration file. Lastly, the client calls
 
 For information about the path resolution, please refer to the [Path Resolution](#path-resolution-and-object-loading) section.
 
+### Enabling Hazelcast Cloud Discovery
+The purpose of Hazelcast Cloud Discovery is to provide clients to use ip addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, you should add a discovery token to the `discoveryToken` field and make true the `enabled` field.
```
Comment (Serdaro) [inline CONFIG.md]: The purpose of Hazelcast Cloud Discovery is to provide clients to use IP addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, specify a token for the `discoveryToken` field and set the `enabled` field to "true".

File: CONFIG.md [hunk 2]
```diff
@@ -291,6 +291,30 @@ the properties section in the JSON configuration file. Lastly, the client calls
...
+        "hazelcastCloud": {
+            "discoveryToken": "EXAMPLE_TOKEN",
+            "enabled": true
+        }
+    }
+}
+
+```
+
+To be able to connect to the provided ip addresses, you should use secure TLS/SSL connection between the client and members. Therefore, you should set an SSL configuration as described in the previous section.
```
Comment (Serdaro) [inline CONFIG.md]: ....provided IP addresses, you should....

File: CONFIG.md [hunk 3]
```diff
@@ -291,6 +291,30 @@ the properties section in the JSON configuration file. Lastly, the client calls
 
 For information about the path resolution, please refer to the [Path Resolution](#path-resolution-and-object-loading) section.
 
+### Enabling Hazelcast Cloud Discovery
+The purpose of Hazelcast Cloud Discovery is to provide clients to use IP addresses provided by `hazelcast orchestrator`. To enable Hazelcast Cloud Discovery, specify a token for the `discoveryToken` field and set the `enabled` field to "true".
```
Comment (mustafaiman) [inline CONFIG.md]: I think what `a token` is is not clear here. Maybe you should express where this token comes from (It is provided by hazelcast orchestrator I assume)

Comment (furkansenharputlu) [reply CONFIG.md]: For now, I don't know how user gets the token. When Hazelcast Cloud is out, I think it becomes clear.

**Pull request merged**